api/v1/contexts/PKPContextHandler.inc.php

```diff
@@ -0,0 +1,490 @@
+			return $response->withStatus(403)->withJsonError('api.contexts.403.contextsDidNotMatch');
+		}
+
+		// A disabled journal can only be access by site admins and users with a
+		// manager role in that journal
+		if (!$context->getEnabled()) {
+			$userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+			if (!in_array(ROLE_ID_SITE_ADMIN, $userRoles)) {
+				$roleDao = DaoRegistry::getDao('RoleDAO');
+				if (!$roleDao->userHasRole($context->getId(), $user->getId(), ROLE_ID_MANAGER)) {
+					return $response->withStatus(403)->withJsonError('api.contexts.403.notAllowed');
+				}
+			}
+		}
+
+		$data = $contextService->getFullProperties($context, array(
+			'request' => $request,
+			'slimRequest' => $slimRequest
+		));
+
+		return $response->withJson($data, 200);
+	}
+
+	/**
+	 * Get the theme and any theme options for a context
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 *
+	 * @return Response
+	 */
+	public function getTheme($slimRequest, $response, $args) {
+		$request = $this->getRequest();
+		$user = $request->getUser();
+
+		$contextService = ServicesContainer::instance()->get('context');
+		$context = $contextService->getContext((int) $args['contextId']);
+
+		if (!$context) {
+			return $response->withStatus(404)->withJsonError('api.contexts.404.contextNotFound');
+		}
+
+		// Don't allow to get one context from a different context's endpoint
+		if ($request->getContext() && $request->getContext()->getId() !== $context->getId()) {
+			return $response->withStatus(403)->withJsonError('api.contexts.403.contextsDidNotMatch');
+		}
+
+		// A disabled journal can only be access by site admins and users with a
+		// manager role in that journal
+		if (!$context->getEnabled()) {
+			$userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+			if (!in_array(ROLE_ID_SITE_ADMIN, $userRoles)) {
+				$roleDao = DaoRegistry::getDao('RoleDAO');
+				if (!$roleDao->userHasRole($context->getId(), $user->getId(), ROLE_ID_MANAGER)) {
+					return $response->withStatus(403)->withJsonError('api.contexts.403.notAllowed');
+				}
+			}
+		}
+
+		$allThemes = PluginRegistry::loadCategory('themes', true);
+		$activeTheme = null;
+		foreach ($allThemes as $theme) {
+			if ($context->getData('themePluginPath') === $theme->getDirName()) {
+				$activeTheme = $theme;
+				break;
+			}
+		}
+
+		if (!$activeTheme) {
+			return $response->withStatus(404)->withJsonError('api.themes.404.themeUnavailable');
+		}
+
+		$data = array_merge(
+			$activeTheme->getOptionValues($context->getId()),
+			['themePluginPath' => $theme->getDirName()]
+		);
+
+		ksort($data);
+
+		return $response->withJson($data, 200);
+	}
+
+	/**
+	 * Add a context
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 *
+	 * @return Response
+	 */
+	public function addContext($slimRequest, $response, $args) {
```

Comment:
I don't see any role checking here... Did I miss it?

Reply:
Have I missed something in the `authorize` method? https://github.com/NateWr/pkp-lib/blob/i3594_forms/api/v1/contexts/PKPContextHandler.inc.php#L75-L89

Reply:
Newp, that's my oversight, thanks!